app/services/dqt/client.rb

```diff
@@ -0,0 +1,26 @@
+module Dqt
+  class Client
+    class Request
+      include HTTParty
+      base_uri ENV.fetch("DQT_BASE_URL", "")
+      headers "Accept" => "application/json",
+              "Content-Type" => "application/json;odata.metadata=minimal",
+              "Authorization" => "Bearer #{ENV.fetch("DQT_API_KEY", "")}",
+              "X-Api-Version" => ENV.fetch("DQT_API_MINOR_VERSION", "20240101")
+    end
+
+    class HttpError < StandardError; end
+
+    GET_SUCCESS = 200
+
+    def self.get(path)
+      response = Request.get(path)
+
+      if response.code == GET_SUCCESS
+        JSON.parse(response.body || "{}")
+      else
+        raise(HttpError, "status: #{response.code}, body: #{response.body}, headers: #{response.headers}")
+      end
+    end
```

Comment:
HTTParty implements some dynamic helpers for checking status codes. This might clean it up a tad:

```suggestion
    def self.get(path)
      response = Request.get(path)

      if response.ok? # Status code == 200 OK - You could also do response.success? which checks for the 2xx range.
        JSON.parse(response.body || "{}")
      else
        raise(HttpError, "status: #{response.code}, body: #{response.body}, headers: #{response.headers}")
      end
    end
```

But furthermore, I'm _slightly_ conscious about the decision to only return the body.

In this specific case, we may have no use of the response headers right now, I'm not sure what headers are returned either. The Ruby net/http _does_ take the approach of `get` for body and `get_response` for the entire response, but something to be aware of. https://docs.ruby-lang.org/en/3.2/Net/HTTP.html#class-Net::HTTP-label-Single-Request+Session

Reply:
I was not aware of the `ok?` method, sounds good, I'll implement that.

I think if we have a use case for wanting more than the body, we can implement it, but until then, I don't see an argument for doing that here.

Reply:
resolved in https://github.com/DFE-Digital/itt-mentor-services/pull/262/commits/738df6523b08d69f96b80c3657b30c3a0214f85d